apps/remix-ide/src/app/files/fileProvider.js

```diff
@@ -71,133 +71,139 @@ class FileProvider {
   async exists (path) {
     // todo check the type (directory/file) as well #2386
     // currently it is not possible to have a file and folder with same path
-    const ret = this._exists(path)
+    const ret = await this._exists(path)
 
     return ret
   }
 
-  _exists (path) {
+  async _exists (path) {
     path = this.getPathFromUrl(path) || path // ensure we actually use the normalized path from here
     var unprefixedpath = this.removePrefix(path)
-    return path === this.type ? true : window.remixFileSystem.existsSync(unprefixedpath)
+    return path === this.type ? true : await window.remixFileSystem.exists(unprefixedpath)
   }
 
   init (cb) {
     cb()
   }
 
-  get (path, cb) {
+  async get (path, cb) {
     cb = cb || function () { /* do nothing. */ }
     path = this.getPathFromUrl(path) || path // ensure we actually use the normalized path from here
     var unprefixedpath = this.removePrefix(path)
-    var exists = window.remixFileSystem.existsSync(unprefixedpath)
-    if (!exists) return cb(null, null)
-    window.remixFileSystem.readFile(unprefixedpath, 'utf8', (err, content) => {
-      cb(err, content)
-    })
+    try {
+      const content = await window.remixFileSystem.readFile(unprefixedpath, 'utf8')
```

Comment:
pointing out that you removed `existsSync(unprefixedpath)` from here, that's maybe fine like that, just making sure you've seen it.